src/main/java/seedu/address/model/step/UniqueStepList.java

```diff
@@ -47,6 +48,19 @@ public void add(Step toAdd) {
         internalList.add(toAdd);
     }
 
+    /**
+     * Adds a step to given index in the list.
+     * The step must not already exist in the list.
+     */
+    public void add(Index index, Step toAdd) {
```

Comment:
Hmm maybe we can change UniqueStepList to OrderedStepList instead, to emphasize that the steps are ordered, and reflect that there are methods to support inserting steps are specific places. Can consider for a future PR.